client/my-sites/email/titan-mail-add-mailboxes/index.jsx

```diff
@@ -0,0 +1,305 @@
+		const new_quantity = mailboxes.length;
+		const email_users = mailboxes.map( transformMailboxForCart );
+
+		return titanMailMonthly( {
+			domain: selectedDomainName,
+			quantity,
+			extra: {
+				email_users,
+				new_quantity,
+			},
+		} );
+	};
+
+	handleContinue = () => {
+		const { selectedSite } = this.props;
+		const { mailboxes } = this.state;
+		const canContinue = areAllMailboxesValid( mailboxes, [ 'alternativeEmail' ] );
+
+		this.recordClickEvent( 'calypso_email_management_titan_add_mailboxes_continue_button_click', {
+			can_continue: canContinue,
+			mailbox_count: mailboxes.length,
+		} );
+
+		if ( canContinue ) {
+			this.props.shoppingCartManager
+				.addProductsToCart( [
+					fillInSingleCartItemAttributes( this.getCartItem(), this.props.productsList ),
+				] )
+				.then( () => {
+					const { errors } = this.props?.cart?.messages;
+					if ( errors && errors.length ) {
+						// Stay on the page to show the relevant error
+						return;
+					}
+					return this.isMounted && page( '/checkout/' + selectedSite.slug );
+				} );
+		}
+	};
+
+	handleUnusedMailboxFinishSetupClick = () => {
+		const {
+			currentRoute,
+			isSelectedDomainNameValid,
+			selectedDomainName,
+			selectedSite,
+		} = this.props;
+
+		this.recordClickEvent( 'calypso_email_management_titan_add_mailboxes_create_mailbox_click' );
+
+		const domainName = isSelectedDomainNameValid ? selectedDomainName : null;
+
+		if ( isEnabled( 'titan/iframe-control-panel' ) ) {
+			page(
+				emailManagementManageTitanAccount( selectedSite.slug, domainName, currentRoute, {
+					context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+				} )
+			);
+
+			return;
+		}
+
+		window.open(
+			emailManagementTitanControlPanelRedirect( selectedSite.slug, domainName, currentRoute, {
+				context: TITAN_CONTROL_PANEL_CONTEXT_CREATE_EMAIL,
+			} )
+		);
+	};
+
+	onButtonClick = () => {
+		this.setState( { quantity: this.state.quantity + 1 } );
+	};
+
+	onMailboxesChange = ( updatedMailboxes ) => {
+		this.setState( { mailboxes: updatedMailboxes, quantity: updatedMailboxes.length } );
+	};
+
+	renderForm() {
+		const { isLoadingDomains, selectedDomainName, titanMonthlyProduct, translate } = this.props;
+
+		if ( isLoadingDomains || ! titanMonthlyProduct ) {
+			return <AddEmailAddressesCardPlaceholder />;
+		}
+
+		return (
+			<>
+				<SectionHeader label={ translate( 'Add New Mailboxes' ) } />
+
+				<Card>
+					<TitanNewMailboxList
+						domain={ selectedDomainName }
+						mailboxes={ this.state.mailboxes }
+						onMailboxesChange={ this.onMailboxesChange }
+					>
+						<Button
+							className="titan-mail-add-mailboxes__action-cancel"
+							onClick={ this.handleCancel }
+						>
+							{ translate( 'Cancel' ) }
+						</Button>
+						<Button
+							className="titan-mail-add-mailboxes__action-continue"
+							primary
+							onClick={ this.handleContinue }
+						>
+							{ translate( 'Continue' ) }
```

Comment:
I understand this is from G Suite's component, but maybe we can be a bit clearer here, like "Purchase x mailboxes" or just "Continue to checkout" / "Proceed to checkout".

Reply:
I'd rather get some consolidated feedback so we can improve both components. I'll leave this for now, as I think it's OK for an initial starting point.